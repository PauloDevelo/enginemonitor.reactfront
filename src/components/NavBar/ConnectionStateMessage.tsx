import React from "react";

import { FormattedMessage, defineMessages } from 'react-intl';

import jsonMessages from "./NavBar.messages.json";
const navBarMsg = defineMessages(jsonMessages);

type ConnectionState = {
    isOnline: boolean,
    isSynced: boolean
};

const ConnectionStateMessage = ({isOnline, isSynced}:ConnectionState) => {
    if(isSynced){
        if (isOnline){
            return <FormattedMessage {...navBarMsg.online} />
        }
        else{
            return <FormattedMessage {...navBarMsg.notonline} />
        }
    }
    else{
        if (isOnline){
            return <FormattedMessage {...navBarMsg.syncAction} />
        }
        else{
            return <FormattedMessage {...navBarMsg.notonline} />
        }
    }
}

export default React.memo(ConnectionStateMessage);