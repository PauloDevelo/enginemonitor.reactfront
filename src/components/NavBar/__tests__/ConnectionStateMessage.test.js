import React from 'react';
import { mount } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import localforage from 'localforage';

import ConnectionStateMessage from '../ConnectionStateMessage';

import ignoredMessages from '../../../testHelpers/MockConsole';

jest.mock('localforage');

describe('ConnectionStateMessage', () => {
  beforeEach(() => {
  });

  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('Could not find required `intl` object.');
  });

  afterEach(async () => {
  });

  const connectionStates = [
    { isOnline: true, isSynced: true },
    { isOnline: true, isSynced: false },
    { isOnline: false, isSynced: true },
    { isOnline: false, isSynced: false },
  ];

  describe.each(connectionStates)('Render', ({ isOnline, isSynced }) => {
    it(`should display the connection when isOnline is ${isOnline} and the application is synced ${isSynced}`, async (done) => {
      // Arrange

      // Act
      const connectionStateMessageWrapper = mount(<ConnectionStateMessage isOnline={isOnline} isSynced={isSynced} />);

      // Assert
      expect(connectionStateMessageWrapper).toMatchSnapshot();
      done();
    });
  });
});
