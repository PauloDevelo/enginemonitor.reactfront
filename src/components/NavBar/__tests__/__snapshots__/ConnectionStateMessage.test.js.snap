// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStateMessage synchronize should display the connection when isOnline is false and the application is synced false 1`] = `
<Memo(ConnectionStateMessage)
  isOnline={false}
  isSynced={false}
>
  <FormattedMessage
    defaultMessage="Disconnected"
    id="notonline"
    values={Object {}}
  >
    Disconnected
  </FormattedMessage>
</Memo(ConnectionStateMessage)>
`;

exports[`ConnectionStateMessage synchronize should display the connection when isOnline is false and the application is synced true 1`] = `
<Memo(ConnectionStateMessage)
  isOnline={false}
  isSynced={true}
>
  <FormattedMessage
    defaultMessage="Disconnected"
    id="notonline"
    values={Object {}}
  >
    Disconnected
  </FormattedMessage>
</Memo(ConnectionStateMessage)>
`;

exports[`ConnectionStateMessage synchronize should display the connection when isOnline is true and the application is synced false 1`] = `
<Memo(ConnectionStateMessage)
  isOnline={true}
  isSynced={false}
>
  <FormattedMessage
    defaultMessage="Synchronize"
    id="syncAction"
    values={Object {}}
  >
    Synchronize
  </FormattedMessage>
</Memo(ConnectionStateMessage)>
`;

exports[`ConnectionStateMessage synchronize should display the connection when isOnline is true and the application is synced true 1`] = `
<Memo(ConnectionStateMessage)
  isOnline={true}
  isSynced={true}
>
  <FormattedMessage
    defaultMessage="Online"
    id="online"
    values={Object {}}
  >
    Online
  </FormattedMessage>
</Memo(ConnectionStateMessage)>
`;
